### PR TITLE
Expand user's home directory in PIPENV_PIPFILE

### PR DESCRIPTION
**Pull request by muelli**
I tried to run pipenv run from another directory. That, however, failed, because it couldn't find my Pipfile. 

### The fix

I went for expanding the tilde in environments.py after having tried to expand it in the [check](https://github.com/pypa/pipenv/blob/c3a9e8d028ae2835ed92ce0318d297d29d856f9c/pipenv/project.py#L100). That failed with a double expansion later. Expanding very early seems to be the cleanest approach to me. 

### The checklist

* [ ] Associated issue
* [ ] A news fragment in the `news/` directory to describe this fix with the extension `.bugfix`, `.feature`, `.behavior`, `.doc`. `.vendor`. or `.trivial` (this will appear in the release changelog). Use semantic line breaks and name the file after the issue number or the PR #.

### CODE REVIEW

File: pipenv/environments.py [hunk 1]
```diff
@@ -157,7 +157,7 @@ def _is_env_truthy(name):
 https://github.com/sindresorhus/cli-spinners
 """
 
-PIPENV_PIPFILE = os.environ.get("PIPENV_PIPFILE")
+PIPENV_PIPFILE = os.path.expaduser(os.environ.get("PIPENV_PIPFILE")) if os.environ.get("PIPENV_PIPFILE") else None
```
Comment (frostming) [inline pipenv/environments.py]: missing 'n' in `expanduser`

File: pipenv/environments.py [hunk 2]
```diff
@@ -157,7 +157,7 @@ def _is_env_truthy(name):
 https://github.com/sindresorhus/cli-spinners
 """
 
-PIPENV_PIPFILE = os.environ.get("PIPENV_PIPFILE")
+PIPENV_PIPFILE = os.path.expanduser(os.environ.get("PIPENV_PIPFILE")) if os.environ.get("PIPENV_PIPFILE") else None
```
Comment (frostming) [inline pipenv/environments.py]: IMO a better place for this code is
https://github.com/pypa/pipenv/blob/3b9b7172293169ad5ce0b7be77e6f27e3dbcde7b/pipenv/project.py#L99-L107

File: pipenv/project.py [hunk 3]
```diff
@@ -98,7 +98,7 @@ def preferred_newlines(f):
 
 if PIPENV_PIPFILE:
     if not os.path.isfile(PIPENV_PIPFILE):
-        raise RuntimeError("Given PIPENV_PIPFILE is not found!")
+        raise RuntimeError("Given PIPENV_PIPFILE (%s) is not found!", PIPENV_PIPFILE)
```
Comment (frostming) [inline pipenv/project.py]: Care to change it to `format` style?

**Comment by body20002** [comment]
could be related to #5532

**Comment by matteius** [comment]
Stale PR -- closing out.

**Pull request closed without merging**